pymech/readma2.py

```diff
@@ -0,0 +1,76 @@
+def readma2(fname,ldim):
+    """A function for reading binary map files (*.ma2) for nek5000.
+
+    The map file comtains, for each element in the mesh, the id of the MPI rank that owns it 
+    followed by the ids of the vertices of the element in the global address space.
+
+    The partitioning is determined by generating the undirected graph formed by the mesh, then 
+    repeatedly computing and deviding the graph using the Fiedler vector of the graph Laplacian. 
+
+    Parameters
+    ----------
+    fname : str
+            file name
+    ldim  : int
+            dimension of the mesh
+    """
+    try:
+        infile = open(fname, "rb")
+    except OSError as e:
+        logger.critical(f"I/O error ({e.errno}): {e.strerror}")
+        return -1
+    #
+    # read header
+    header   = infile.read(132).split()
+    nel      = int(header[1])
+    nactive  = int(header[2])
+    depth    = int(header[3])
+    d2       = int(header[4])
+    npts     = int(header[5])
+    nrnk     = int(header[6])
+    noutflow = int(header[7])
```

Comment:
You don't have to use cryptic variable names. Longer variable names would be nice to have here (and comments if it is still unclear).

Reply:
Most of these are not really used, but since I started parsing the header I thought I'd add all data there. I have added comments but kept the naming since it's not as cryptic as you make it sound, especially regarding the values that are being used in this function.

Reply:
I see. In that case leave a comment, so we know read more about it in Nek5000's code in the future.